### PR TITLE
CLDR-15359 build: always build exhaustive

### PR DESCRIPTION
**Pull request by srl295**
CLDR-15359

This only makes sure cldr-archive is present before mvn tests

- [ ] This PR completes the ticket.

ALLOW_MULTI_COMMITS=true

### CODE REVIEW

**Comment by srl295** [comment]
@macchiati  this doesn't explicitly change any test code or behavior.

It does add a couple of minutes to the build, because usually the fetch only gets one commit instead of all of them, plus the time to process the checkout of all of the tags.

**Comment by srl295** [comment]
we could merge this, but may not be worth it unless we use it for tests. 

Is there a way we could make the tests optionally run without requiring cldr-archive to be present for all test runs?

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -17,11 +17,13 @@ jobs:
         uses: actions/checkout@v3
         with:
           lfs: false
+          fetch-depth: 0  # Needed for cldr-archive
```
Comment (btangmu) [inline .github/workflows/maven.yml:20]: I'm not sure what this means...

I found this, not sure what the implication is:
https://github.com/changesets/action/issues/97 "This is no longer necessary: the tool will automatically deepen shallow clones as necessary"

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -52,6 +54,9 @@ jobs:
           -DskipTests=true
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Checkout CLDR archve
```
Comment (btangmu) [inline .github/workflows/maven.yml:57]: typo "archve" --> "archive"

**Review by btangmu — DISMISSED** [review]

File: .github/workflows/maven.yml (hunk 1 above)
Comment (srl295) [reply .github/workflows/maven.yml:20]: I think that's for an unrelated tool. cldr-archive does _not_ automatically deepen shallow clones as necessary.

File: .github/workflows/maven.yml (hunk 2 above)
Comment (srl295) [reply .github/workflows/maven.yml:57]:
```suggestion
      - name: Checkout CLDR archive
```

File: .github/workflows/maven.yml (hunk 1 above)
Comment (btangmu) [reply .github/workflows/maven.yml:20]: good to know! so I'm guessing zero means no limit

**Review by btangmu — APPROVED** [review]

**Review by macchiati — APPROVED** [review]

**Pull request merged**